### ISSUE TITLE
App Configuration Policies-Settings details 

### ISSUE THREAD

**Issue by eDucatehealth** [Micke-K/IntuneManagement#231]
In the App Configuration policies (referred to as App Configuration (Devices) in Intune Manager), the 'Settings' details are not being retrieved for Android devices, whereas they are successfully fetched on iOS. Any suggestions?

**Comment by Micke-K** [comment]
Hello,

I tried to test this in my new test environment but I don't have a Google Enterprise account so I can't setup the Google Store connection. 

Anything in the log? 

Cheers!

**Comment by EUC-Guy** [comment]
Hi,
I have the same issue.
Entry in logs.
Warning	No category files returned for AndroidForWorkOemConfig

**Comment by Micke-K** [comment]
Thank you for this. I'll see if I can figure something out based on this. 

Cheers!

**Comment by Micke-K** [comment]
Hello,

I had a quick look. That log warning is a part of the documentation. Is this issue about the documentation is missing info? 

Just trying to make sure I'm looking at the right thing. 

Cheers!

**Comment by eDucatehealth** [comment]
Yes, the 'Settings' details are not being retrieved for Android devices.

**Comment by Micke-K** [comment]
Hello,

Can you test this? I don't have any Android apps in my test environment so I can't create an Android policy.

I basically copied the iOS documentation and did the same for Android.

Let me know how it goes. Check the log if it is not working.

Cheers!

[DocumentationCustom.zip](https://github.com/user-attachments/files/15856462/DocumentationCustom.zip)

**Comment by eDucatehealth** [comment]
Appreciate your assistance and time. I checked and still same result. 
Checked the Log and showing 
“Warning: No category files returned for AndroidForWorkOemConfig”

**Comment by Micke-K** [comment]
Hello,

Hmm, I was expecting it to at least try to document it now, instead of fall back to default documentation.

Can you upload a json file of the policy, so I have something to test with? Or create a new one with dummy values.

Cheers!

**Comment by Micke-K** [comment]
Hello

Can you try this one? This should support AndroidForWorkOemConfig profiles. At least I hope so. I don't have one in the environment but was able to swap object during the documentation and this worked.

![image](https://github.com/Micke-K/IntuneManagement/assets/43226266/7eb9ce5d-f01a-4620-b126-70a9738882ef)

Cheers!

[DocumentationCustom.zip](https://github.com/user-attachments/files/15942804/DocumentationCustom.zip)

**Comment by eDucatehealth** [comment]
Thanks. Sorry, I'm not very good with scripting, which is why it's taking me some time to generate the JSON file. I tried your script, and for some Android policies, it is returning the "Value output." However, when using it for MS Defender policies, it doesn't provide the output for "Value Type" and "Configuration Value," although it does give the output for "Configuration Key." Additionally, for both iOS and Android, it is not providing any output for the "Assignments" of the policy. I've attached two screenshots for reference: one showing the actual policy and the other showing the policy output. Thanks again for all your assistance.
![App_Device_MS-defender_Screenshot](https://github.com/Micke-K/IntuneManagement/assets/83241939/32df5cd7-ad77-4c0e-9a57-47d9dc9c55b2)
------
![Output_App_Device_MS-defender_Screenshot](https://github.com/Micke-K/IntuneManagement/assets/83241939/8cd41aaf-93a1-4f4f-965c-48fd359ee825)

**Comment by Micke-K** [comment]
We're getting closer. I'll have a look and see if I can figure it out. I think I know why values are not showing up. 

First step is to get key and value. Changing that to UI text might be possibly but most likely lots of work.

Cheers!

**Comment by eDucatehealth** [comment]
Bingo… Thanks, it worked! Although the "Value Type" output isn't showing, it’s a constant value, so its absence doesn’t make much difference. Attached is the new output screenshot. Thanks again! 

![Revised_Output_App_Device_MS-defender_Screenshot](https://github.com/Micke-K/IntuneManagement/assets/83241939/8905024c-8cba-427c-8574-52863e2ecded)
